src/darwin/mod.rs

```diff
@@ -277,26 +277,30 @@ static FONT_SMOOTHING_ENABLED: Lazy<bool> = Lazy::new(|| {
         let key = NSString::alloc(nil).init_str("AppleFontSmoothing");
         let value: id = msg_send![id::standardUserDefaults(), objectForKey: key];
 
-        if !msg_send![value, isKindOfClass: class!(NSNumber)] {
-            return true;
-        }
+        if msg_send![value, isKindOfClass: class!(NSNumber)] {
+            let num_type: id = msg_send![value, objCType];
+            if num_type == nil {
+                return true;
+            }
 
-        let num_type: id = msg_send![value, objCType];
-        if num_type == nil {
-            return true;
-        }
+            // NSNumber's objCType method returns one of these strings depending on the size:
+            // q = quad (long long), l = long, i = int, s = short.
+            // This is done to reject booleans, which are NSNumbers with an objCType of "c", but
+            // macOS does not treat them the same as an integer 0 or 1 for this setting,
+            // it just ignores it.
+            let int_specifiers: [&[u8]; 4] = [b"q", b"l", b"i", b"s"];
+            if !int_specifiers.contains(&CStr::from_ptr(num_type as *const i8).to_bytes()) {
+                return true;
+            }
 
-        // NSNumber's objCType method returns one of these strings depending on the size:
-        // q = quad (long long), l = long, i = int, s = short.
-        // This is done to reject booleans, which are NSNumbers with an objCType of "c", but macOS
-        // does not treat them the same as an integer 0 or 1 for this setting, it just ignores it.
-        let int_specifiers: [&[u8]; 4] = [b"q", b"l", b"i", b"s"];
-        if !int_specifiers.contains(&CStr::from_ptr(num_type as *const i8).to_bytes()) {
-            return true;
+            let smoothing: id = msg_send![value, integerValue];
+            return smoothing as i64 != 0;
+        } else if msg_send![value, isKindOfClass: class!(NSString)] {
+            let smoothing: i64 = msg_send![value, intValue];
```

Comment:
you should use `integerValue` and do basically the same as code above. with checking.

Reply:
I just read the docs https://developer.apple.com/documentation/foundation/nsstring/1414988-intvalue?language=objc

Telling to use integeterValue. Nothing more.

Reply:
Sorry, I was confused by the previous implementation storing an `NSInteger` in an `id` so I thought it was a ref-counted type.

Reply:
Similarly the `num_type` is also using the wrong type, it should be:
```
 let num_type: *const c_char = msg_send![value, objCType];
if num_type.is_null() {
    return true;
}
```
Should I include fix in PR?

Reply:
I guess. Also, once you fixed something you should re-request review, otherwise I'm likely to miss it.